scripts/o3de/ExportScripts/export_source_android.py

```diff
@@ -0,0 +1,218 @@
+                           deploy_to_device: bool,
+                           org_name: str|None,
+                           activity_name:str|None,
+                           logger: logging.Logger|None) -> None:
+    o3de_cli_script_path = ctx.engine_path / ('scripts/o3de' + android_support.O3DE_SCRIPT_EXTENSION)
+
+    if not logger:
+        logger = logging.getLogger()
+        logger.setLevel(logging.ERROR)
+
+    is_installer_sdk = manifest.is_sdk_engine(engine_path=ctx.engine_path)
+
+    def run_android_command(args, cwd, error_msg):
+        has_errors = exp.process_command(args, cwd=cwd)
+        if has_errors:
+            raise exp.ExportProjectError(error_msg)
+    
+    run_android_command([o3de_cli_script_path, 'android-configure', '--validate'], cwd=ctx.project_path, error_msg="Invalid Android Configuration")
+
+    # Calculate the tools and game build paths
+    default_base_path = ctx.engine_path if engine_centric else ctx.project_path
+
+    # Resolve (if possible) and validate any provided seedlist files
+    validated_seedslist_paths = exp.validate_project_artifact_paths(project_path=ctx.project_path,
+                                                                    artifact_paths=seedlist_paths)
+    
+    # Preprocess the seed file paths in case there are any wildcard patterns to use
+    preprocessed_seedfile_paths = exp.preprocess_seed_path_list(project_path=ctx.project_path,
+                                                                paths=seedfile_paths)
+    
+    eutil.process_level_names(ctx, preprocessed_seedfile_paths, level_names, ASSET_PLATFORM)
+
+    tools_build_path = eutil.handle_tools(ctx, should_build_tools, is_installer_sdk, tool_config, '', tools_build_path, default_base_path, engine_centric, logger)
+
+    eutil.handle_assets(ctx, should_build_all_assets, tools_build_path, is_installer_sdk, 
+                        tool_config, engine_centric, fail_on_asset_errors,
+                        ASSET_PLATFORM, ASSET_PLATFORM, validated_seedslist_paths, preprocessed_seedfile_paths, 
+                        ctx.project_path/'AssetBundling', max_bundle_size,
+                        logger)
+
+    android_project_config= android_support.get_android_config(project_path=None)
+
+    android_sdk_home = pathlib.Path(android_project_config.get_value(key=android_support.SETTINGS_SDK_ROOT.key))
+
+    run_android_command([o3de_cli_script_path, 'android-generate', '-p', ctx.project_name, '-B', target_android_project_path, "--asset-mode", asset_pack_mode],
+                        cwd=default_base_path,
+                        error_msg="Android Project Generation Failure.")
+
+    run_android_command([target_android_project_path / ('gradlew'+ android_support.GRADLE_EXTENSION), f'assemble{build_config.title()}'],
+                        cwd=target_android_project_path,
+                        error_msg="Gradle Build Failure.")
+    
+    if deploy_to_device:
```

Comment:
In order to deploy to a device, the android project generation must have the keystore file, passwords, etc configured when the android project generation is done. This is optional by itself with the android tools, because you can manually open the android project with android studio and add it there. But if this is a workflow purely based on using the command line workflow, you will need to check if the generated gradle build script has this setting. 
Take a look at the generated `build.gradle` file for `app` and see if the `signingConfigs` section is specified. (under `target_android_project_path/app/build.gradle`.  It should look something like:

```
//
// Copyright (c) Contributors to the Open 3D Engine Project.
// For complete copyright and license terms please see the LICENSE at the root of this distribution.
//
// SPDX-License-Identifier: Apache-2.0 OR MIT
//

apply plugin: "com.android.application"

android {
    namespace "org.o3de.Paper_Kid"

    signingConfigs {
        debug {
            storeFile file('C:/Users/tjkotha/o3de.keystore')
            storePassword 'o3de_android'
            keyPassword 'o3de_android'
            keyAlias 'o3de_key'
        }
        profile {
            storeFile file('C:/Users/tjkotha/o3de.keystore')
            storePassword 'o3de_android'
            keyPassword 'o3de_android'
            keyAlias 'o3de_key'
        }
        release {
            storeFile file('C:/Users/tjkotha/o3de.keystore')
            storePassword 'o3de_android'
            keyPassword 'o3de_android'
            keyAlias 'o3de_key'
        }
    }

    compileSdkVersion sdkVer
...
```
If this doesn't exist, then the deployment will fail because the APK won't be signed.

Reply:
does the android configure validate command check for keystore file, passwords, etc configured, or does that only occur after the project is generated?

Reply:
No, because it is meant to be optional as part of the android-generate command. Android generate doesn't whether you use command line or Android Studio to build the project. The APK does not have to be signed if you intend to use an emulator, for example, which Android Studio provides. And Android Studio can add signing configs to the existing android project anyways.